File: InJava/app/src/main/java/com/barryzeha/quizquestion/model/database/RankingDao.java
package com.barryzeha.quizquestion.model.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.barryzeha.quizquestion.model.RankingModel;

import java.util.List;

/****
 * Project QuizQuestion
 * Created by Barry Zea H. on 01/10/2022
 * Copyright (c)  All rights reserved.
 ***/
@Dao
public
interface RankingDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveRanking(RankingModel ranking);
    @Query("select * from  ranking_table order by points desc limit 10")
    List<RankingModel> getRanking();
}

File: InKotlin/app/src/main/java/com/barryzeha/quizquestion/model/database/RankingDAO.kt
package com.barryzeha.quizquestion.model.database

import androidx.room.*
import com.barryzeha.quizquestion.model.RankingModel


/****
 * Project QuizQuestion
 * Created by Barry Zea H. on 27/09/2022
 * Copyright (c)  All rights reserved.
 ***/
@Dao
interface RankingDAO {
@Insert(onConflict = OnConflictStrategy.REPLACE)
fun saveRanking(ranking: RankingModel)
@Query("select  * from ranking_table order by points desc")
fun getRankings():MutableList<RankingModel>

}